include/deal.II/grid/grid_tools.h

```diff
@@ -983,6 +987,73 @@ namespace GridTools
   get_patch_around_cell(const typename Container::active_cell_iterator &cell);
 
 
+  /**
+   * This function takes a vector of active cells (here named as patch_cells) 
+   * as input argument, and returns vector of their parent cells with the 
+   * coarsest common level of refinement.
+   * The way that the function works, is as follows:
+   * first it computes the minimum refinement level of the given vector of
+   * active cells. Then loops over cells of the input vector of active cells.
+   * For each cell, if its refinement level is equal to the computted minimum 
+   * refinement level, that cell is pushed back to the vector of output cells.
+   * Otherwise, the function looks for the parent cell of the current cell, 
+   * with the level equal to the computed minimum refinement level, and 
+   * then pushes back this parent cell to the vector of output cells.
+   *
+   * @tparam Container
+   * In C++, the compiler can not determine the type of <code>Container</code>
+   * from the function call. You need to specify it as an explicit template
+   * argument following the function name.
+   * @param[in] patch_cells This vector of cells, typically results from 
+   * calling the function GridTools::get_patch_around_cell().
+   * @return A list of cells with the coarsest common level of refinement
+   *
+   * @author Arezou Ghesmati, Wolfgang Bangerth, 2015
+   */
+  template <class Container>
+  std::vector<typename Container::cell_iterator> get_cells_at_coarsest_common_level(
+    const std::vector<typename Container::active_cell_iterator> &patch_cells);
+
+  /**
+   * The first element of input argument is a vector of active cells which we
```

Comment:
@Arezou-gh -- Timo knows this, but the average user won't, so he wants you to put this kind of explanation into the documentation :-)

When you write documentation, you need to put yourself into the shoes of a potential reader. If you were reading about a function of which you don't know what it does, you would ask yourself the kind of questions Timo asked you. Use this perspective -- of thinking what a reader would want to know about a function -- when you write a function. In particular, the user doesn't actually care about *how the function is implemented*. The important point is *what the function does*, and what its requirements are on the input parameters.